addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/discovery/NetworkDiscoveryService.java

```diff
@@ -51,15 +78,71 @@ protected void startScan() {
         if (executorService != null) {
             stopScan();
         }
-
-        logger.debug("Starting Discovery");
+        logger.trace("Starting Discovery");
         LinkedHashSet<String> networkIPs = NetworkUtils.getNetworkIPs(NetworkUtils.getInterfaceIPs());
         scanningNetworkSize = networkIPs.size();
         executorService = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors() * 10);
 
         for (Iterator<String> it = networkIPs.iterator(); it.hasNext();) {
             final String ip = it.next();
-            executorService.execute(new PingRunnable(ip, this));
+            InetAddress destination;
+            try {
+                destination = InetAddress.getByName(ip);
+            } catch (UnknownHostException unknownHostException) {
+                logger.trace("Skip IP that cannot be converted to a InetAddress", unknownHostException);
+                continue;
+            }
+
+            // ARP ping. Use an own executor for each network interface
+            if (canUseARPPing) {
+                Set<String> interfaceNames = NetworkUtils.getInterfaceNames();
+                for (final String interfaceName : interfaceNames) {
+                    executorService.execute(() -> {
+                        try {
+                            NetworkUtils.wakeUpIOS(destination);

```

Comment:
It doesn't harm to send a udp packet before sending an arp packet. But you are right, the method `isIOSdevice()` should be used before calling wakeUpIOS().